software/nosql/src/main/java/brooklyn/entity/nosql/redis/RedisClusterImpl.java

```diff
@@ -72,12 +120,12 @@ public void restart() {
         throw new UnsupportedOperationException();
     }
 
-    protected boolean calculateServiceUp() {
-        boolean up = false;
-        for (Entity member : slaves.getMembers()) {
-            if (Boolean.TRUE.equals(member.getAttribute(SERVICE_UP))) up = true;
-        }
-        return up;
-    }
+    private boolean calculateServiceUp() {
+        if (!Boolean.TRUE.equals(getMaster().getAttribute(SERVICE_UP))) return false;
 
+        for (Entity member : getSlaves().getMembers())
```

Comment:
You can check the status of the `getSlaves()` entity directly, it will update based on the members state. It's at [least one](https://github.com/apache/incubator-brooklyn/blob/master/core/src/main/java/brooklyn/entity/group/DynamicClusterImpl.java#L171) by default.